packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -196,12 +196,19 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
                     itemId: resolvedUrl.itemId,
                 }).then((tokenResponse) => {
                     const token = tokenFromResponse(tokenResponse);
-                    event.end({ fromCache: isTokenFromCache(tokenResponse), isNull: token === null ? true : false });
+                    // This event alone generates so many events that is materially impacts cost of telemetry
+                    // Thus do not report end event when it comes back quickly.
+                    // Note that most of the hosts do not report if result is comming from cache or not,
+                    // so we can't rely on that here
+                    if (event.duration >= 32) {
```

Comment:
Just curious, why 32?

Reply:
It's really a random number :)
We want it low to have better options, and high to have smaller COGS
Based on current numbers, it cuts most of the cost. But any number in range of 20-50 would generally work